src/oemof_tabular_plugins/datapackage/post_processing.py

```diff
@@ -0,0 +1,828 @@
+import logging
+
+import pandas as pd
+from datapackage import Package
+import oemof.solph as solph
+import numpy as np
+
+# ToDo: check to see if the storage optimized input/output (invest_out) and
+#  optimized capacity (invest) are saved correctly
+# ToDo: is another raw output from the results is investment costs? or does this have to be calculated?
+RAW_OUTPUTS = ["investments"]
+PROCESSED_RAW_OUTPUTS = ["flow_min", "flow_max", "aggregated_flow"]
+RAW_INPUTS = [
+    "marginal_cost",
+    "carrier_cost",
+    "capacity_cost",
+    "storage_capacity_cost",
+    "capacity",
+    "capacity_potential",
+    "capacity_minimum",
+    "expandable",
+    "storage_capacity",
+    "storage_capacity_potential",
+    "efficiency",
+    "capex",
+    "opex_fix",
+    "lifetime",
+    "renewable_factor",
+    "emission_factor",
+    "land_requirement_factor",
+    "water_footprint_factor",
+]
+
+
+# Functions for results per component
+def compute_capacity_total(results_df):
+    # ToDo: how it is here is that now total capacity is considering the storage capacity (in MWh) for storage.
+    #  Check how the storage capacities should be displayed in the results to make it not confusing for the user. Maybe
+    #  the storage components need two total capacity results (one for power and one for energy)?
+    """Calculates total capacity by adding existing capacity (capacity) to optimized capacity (investments)"""
+    if "storage" in results_df.name:
+        return results_df.storage_capacity + results_df.investments
+    else:
+        return results_df.capacity + results_df.investments
+
+
+def compute_annuity_total(results_df):
+    """Calculates total annuity by multiplying the annuity by the optimized capacity"""
+    # ToDo: now storage_capacity_cost is used for the annuity if the component is storage.
+    #  Check that this is correctly applied for storage components or if two different costs should
+    #  be calculated (one for power and one for energy)
+    if "storage" in results_df.name:
+        return results_df.storage_capacity_cost * results_df.investments
+    else:
+        return results_df.capacity_cost * results_df.investments
+
+
+def compute_upfront_investment_costs(results_df):
+    # ToDo: check for storage if investments is based on correct parameter
+    """Calculates investment costs by multiplying capex with optimized capacity (investments)"""
+    if "capex" not in results_df.index:
+        return None
+    else:
+        return results_df.capex * results_df.investments
+
+
+def compute_opex_fix_costs(results_df):
+    """Calculates yearly opex costs by multiplying opex with optimized capacity (investments)"""
+    if "opex_fix" not in results_df.index:
```

Comment:
normally this check is not needed as "opex_fix" is listed as a function argument